### PR TITLE
Upgrade react to 0.14 and the rest of outdated dependencies.

### PR DESCRIPTION
**Pull request by rcorral**
* Upgrade react from 0.13 to 0.14
* Removes shrinkwrap
* Deps webpack@1.12.11
* Switching our dependencies to only use latest patch release
* Deps gulp-autoprefixer@3.1.0
* Deps eslint-plugin-react@3.15.0
* Deps glob@6.0.4
* Deps gulp-sourcemaps@1.6.0
* Changed canvas canvas peerdep requirement.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: docs/styles/page-content.less [hunk 1]
```diff
@@ -112,7 +112,7 @@ Page Content
 
           &:before {
 
-            .opacity(100);
+            opacity: 100;
```
Comment (jfurrow) [inline docs/styles/page-content.less]: This should be `opacity: 1`

File: docs/styles/page-content.less [hunk 2]
```diff
@@ -153,7 +153,7 @@ Page Content
...
```
Comment (jfurrow) [inline docs/styles/page-content.less]: Same, `opacity: 11

File: docs/styles/page-content.less (hunk 1 above)
Comment (rcorral) [inline docs/styles/page-content.less]: Done.

File: docs/styles/page-content.less (hunk 2 above)
Comment (rcorral) [inline docs/styles/page-content.less]: Done.

**Comment by jfurrow** [comment]
This is awesome. Good work!

**Comment by jfurrow** [comment]
I'm pretty sure Babel isn't working properly:

File: package.json [hunk 3]
```diff
@@ -71,8 +78,7 @@
     "dist-src": "cp -r ./src ./lib && find ./lib -name '__tests__' -type d -exec rm -r '{}' + && ./node_modules/.bin/babel lib --out-dir lib",
     "livereload": "NODE_ENV='development' ./node_modules/.bin/gulp docs:livereload",
     "npm-publish": "npm publish ./",
-    "test": "mv .babelrc .babelrc-tmp && jest && mv .babelrc-tmp .babelrc",
-    "shrinkwrap": "npm-shrinkwrap --dev",
+    "test": "jest",
```
Comment (kennyt) [inline package.json]: nice

**Comment by kennyt** [comment]
:ship:

**Comment by daiyiwhale** [comment]
:sparkles:

**Comment by daiyiwhale** [comment]
The dawn of a new era

**Pull request merged**